### PR TITLE
Add support for the Pagerduty Events API version 2

### PR DESCRIPTION
**Pull request by orien**
In #64, we applied a refactoring to support multiple Pagerduty Events API versions.

In this PR, we add support for the Events API version 2.

I've reworked @reidcooper's proposal in #52 to work with the new `Pagerduty.build` method.

  ```ruby
  pagerduty = Pagerduty.build(
    integration_key: "<integration-key>",
    api_version:     2
  )
  incident = pagerduty.trigger(
    summary:  "summary",
    source:   "source",
    severity: "critical"
  )
  incident.acknowledge
  incident.resolve
  ```

### CODE REVIEW

File: spec/pagerduty/events_api_v2_spec.rb [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+        Given {
+          allow(transport).to receive(:send_payload).and_return(
+            "status"    => "success",
+            "dedup_key" => "incident-key-in-response",
+            "message"   => "Event processed",
+          )
+        }
+        When(:incident) { pagerduty.trigger(simple_incident_details) }
+        Then { expect(incident).to be_a Pagerduty::EventsApiV2::Incident }
+        Then { incident.incident_key == "incident-key-in-response" }
```
Comment (johnsyweb) [inline spec/pagerduty/events_api_v2_spec.rb:169]: Should this be an expectation?

```suggestion
        Then { expect(incident.incident_key).to eq "incident-key-in-response" }
```

File: spec/pagerduty/events_api_v2_spec.rb [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+    end
+  end
+
+  describe "#incident" do
+    When(:incident) { pagerduty.incident(incident_key) }
+
+    context "a valid incident_key" do
+      Given(:incident_key) { "a-test-incident-key" }
+      Then { expect(incident).to be_a Pagerduty::EventsApiV2::Incident }
+      Then { incident.incident_key == incident_key }
```
Comment (johnsyweb) [inline spec/pagerduty/events_api_v2_spec.rb:201]: _ditto_

File: spec/pagerduty/events_api_v2_spec.rb (hunk 1 above)
Comment (orien) [reply spec/pagerduty/events_api_v2_spec.rb:169]: This is actually the standard way to use `Then` blocks in [rspec-given](https://github.com/rspec-given/rspec-given). As I understand, the block should always return `true`. There is an explanation in the [`Then` documentation](https://github.com/rspec-given/rspec-given#then)

In this instance, if I change code under test to not set the `incident_key` this test fails with the following output:

```
Failures:

  1) Pagerduty#trigger handling responses PagerDuty successfully creates the incident         Then { incident.incident_key == "incident-key-in-response" }

     Failure/Error: Then { incident.incident_key == "incident-key-in-response" }
     
       Then expression failed at /Users/orien/src/pagerduty/spec/pagerduty/events_api_v2_spec.rb:169
       expected: nil
       to equal: "incident-key-in-response"
         false   <- incident.incident_key == "incident-key-in-response"
         nil     <- incident.incident_key
         #<Pagerduty::EventsApiV2::Incident:0x00007fbf20a7c8a8 @integration_key="a-test-routing-key", @incident_key=nil, @transport=#<InstanceDouble(Pagerduty::HttpTransport) (anonymous)>>
                 <- incident
     # ./spec/pagerduty/events_api_v2_spec.rb:169:in `block in Then'
```

It's pretty awesome how it breaks down the comparison so you can see different values in the expression. Pretty handy for debugging.

Comment (johnsyweb) [reply spec/pagerduty/events_api_v2_spec.rb:169]: TIL.

**Pull request merged**